### PR TITLE
Added functionaility to upload videos to youtube

### PR DESCRIPTION
**Pull request by joel-labbe**
Videos do not get auto added to playlists yet, I am looking into why not. I had to change a good amount of the imports as some were merged into others. Also, some Python 2.7 code that was preventing compilation was changed. No new tests were added. In the future if anyone implements this don't do it through putty as you have to give permission to your oauth2 application to access your Google account through a web browser that pops up.

@FanchenBao should we allow people to see our client secret as it is kind of sensitive info? Since people showed just get their own oauth2 app permissions from Google, and not use ours. I think the worst they can do is use up all of our requests or report our application for misuse. Getting a new one is tedious, but not too difficult since it is not verified by Google yet.

### CODE REVIEW

**Comment by FanchenBao** [comment]
@joel-labbe Build failed because of `pkg-resources`. Please remove it from your `requirements.txt`.

Concerning the oauth2 authentication. We can ignore that problem for demo purpose. However, for production, what we can do is that after customer registers their account with us, we provide them with a youtube account and the associated oauth2 authentication. User should not have to worry about all this thing. We can simply provide them the oauth2 file and ask them to put it in a folder and that's it. If the oauth2 expires, we provide them a new one. Concerning using up all the requests, if we are in production, we will probably negotiate a paid account with youtube for each customer such that there will be no limit to the number of requests.

File: requirements.txt [hunk 1]
```diff
@@ -1,39 +1,55 @@
...
 importlib-metadata==0.23
 more-itertools==7.2.0
 nodeenv==1.3.3
 nose==1.3.7
 numpy==1.17.3
+oauth2client==4.1.3
 packaging==19.2
 paho-mqtt==1.4.0
+picamera==1.13
+pkg-resources==0.0.0
```
Comment (FanchenBao) [inline requirements.txt]: Remove `pkg-resources`

Comment (joel-labbe) [reply requirements.txt]: I will get on it soon.

Comment (joel-labbe) [reply requirements.txt]: Okay the change was made.

Comment (FanchenBao) [reply requirements.txt]: Please remove `picamera` as it cannot be installed by Travis's virtual machine. Keep an eye for these packages that usually fail Travis build, e.g. `picamera`, `RPi.GPIO`, `pkg-resources`, etc. I think at the current stage of our project, doing `pip freeze > requirements.txt` probably is no longer a recommended practice.

Comment (FanchenBao) [reply requirements.txt]: My suggestion is first sync your `requirements.txt` with master. Then just add these packages manually `httplib2`, `google-api-python-client`, `oauth2client`. All the other packages are dependencies for these three and will be installed automatically. So there is no need to explicitly add them to `requirements.txt`

Comment (FanchenBao) [reply requirements.txt]: Of course, if adding just these three packages is not enough, you can manually add more if needed.

Comment (joel-labbe) [reply requirements.txt]: So, remove all the new packages added to requirements.txt besides httplib2, google-api-python-client, oauth2client? Or just remove picamera?

Comment (FanchenBao) [reply requirements.txt]: try remove all the new packages except httplib2, google-api-python-client, oauth2client. Then delete your local venv, recreate it, and install with requirements. this is to see if using just these three would be sufficient for your camera interface,

Comment (joel-labbe) [reply requirements.txt]: Did what you asked.

Comment (FanchenBao) [reply requirements.txt]: `test_five` in `test_camera_interface` failed. Here is the error message.

```
_____________________________ TestClass.test_five ______________________________
self = <src.raspberry_pi_camera.tests.test_camera_interface.TestClass object at 0x7f92002c13c8>
    def test_five(self):
        capture = str(sample)
>       assert (
            capture
            == "Video Length: 30, Resolution: (640, 480), Save Location: /home/pi/Videos/"
        )
E       AssertionError: assert 'Video Length...JNiyHgG02ECXN' == 'Video Length...me/pi/Videos/'
E         Skipping 62 identical leading characters in diff, use -v to show
E         - /pi/Videos/, YouTube Livestream Link: https://youtu.be/t48LW4J8b4A, YouTube Playlist Link: https://www.youtube.com/playlist?list=PLTdMMnsiEwSnKNWdLlAEJNiyHgG02ECXN
E         + /pi/Videos/
src/raspberry_pi_camera/tests/test_camera_interface.py:33: AssertionError
========================= 1 failed, 17 passed in 5.57s =========================
The command "python3 -m pytest --ignore=src/raspberry_pi_ui/tests" exited with 1.
```
Seems to me that your capture now returns actual youtube link which is not included in the assertion check.

Has your `pytest` passed locally?

Comment (FanchenBao) [reply requirements.txt]: I checked your branch and saw that you changed the return value of `__str__()` of your class. Yet such change was not reflected in your test file. So you need to update your test case.

Comment (joel-labbe) [reply requirements.txt]: My bad, made the change.

**Pull request merged**